### DISCUSSION TITLE
Disable screensaver / blank screen

### DISCUSSION THREAD

**Discussion by ericemmrich** [neutrinolabs/xrdp#2264] [Q&A] [answered]
Hey everyone,

whenever I connect through xRDP, the screen blanks out after a while even with my system's screensaver and power management options disabled.

When I execute these commands manually, the screen doesn't blank out anymore:

```
xset s off
xset s noblank
```

However, they will only work until I reboot. Afterwards, I need to execute them again.

So I'm looking  for a more permanent solution. Either through a configuration option or by executing these commands automatically whenever I log in. Already tried to place them in `~/.xsession` and `~/.profile` but that did not work.

Any help is appreciated. Many thanks in advance.

**Comment by matt335672** [comment, marked as answer]
Hi @ericemmrich 

There's a simple answer and a complicated answer to this I suspect!

A couple of questions for you:-
- What system/desktop are you running?
- Is this related to unlocking the screen at all?

Thanks.

**Reply by ericemmrich** [reply]
Hey @matt335672,

thank you for the fast reply.

I'm on Zorin OS Lite latest (Ubuntu derivative). Screen Locking as well as Screen Saver and Power Management options are all disabled. Still the screen blanks out after a few minutes of inactivity unless I execute the commands stated. I do not need to unlock the screen after it blanked out, though.

Does that help?

**Reply by matt335672** [reply]
Yes that's fine.

There's a reconnect script `/etc/xrdp/reconnectwm.sh` which runs when a client connects to xrdp. It's got enough context (i.e. a DISPLAY) to be able to run `xset s off`. That would unblank the screen when you reconnect, but still blank the screen when you're not about. In an office situation that might be preferable to disabling the screen blank entirely.

Alternatively:-
1) Add the commands to `/etc/xrdp/startwm.sh`.
1) Add a scriptlet like this to  `/etc/X11/Xsession.d`. There are plenty of examples in there which you can use as a template.

```sh
# Disable screen blanking for xrdp sessions
if [ -n "$XRDP_SESSION"]; then
    xset s off
    xset s noblank
fi
```

**Reply by ericemmrich** [reply]
Many thanks, @matt335672! 

With your help, I was finally able to disable the screen blank out permanently with a slightly modified version of your scriptlet:

```bash
$ cat /etc/X11/Xsession.d/99custom_disable-blank-screen
# Disable screen blanking for xrdp sessions
if [ -n "$XRDP_SESSION" ]; then
    xset s off
    xset s noblank
fi
```

There was just a space missing before the closing `]` which threw an error when executed in a bash.

Also, the file name which determines the execution order was rather important. On my system, the scriptlet needs to execute before `/etc/X11/Xsession.d/99x11-common_start`. Otherwise, it was not executed.

**Reply by matt335672** [reply]
Thanks for fixing my deficient script! Glad you got it working.
